internal/api/external.go

```diff
@@ -44,6 +47,25 @@ func (a *API) ExternalProviderRedirect(w http.ResponseWriter, r *http.Request) e
 	query := r.URL.Query()
 	providerType := query.Get("provider")
 	scopes := query.Get("scopes")
+	flowType := query.Get("flow_type")
+	codeChallengeMethodParam := query.Get("code_challenge_method")
+	var codeChallengeMethod models.CodeChallengeMethod
+
+	switch codeChallengeMethodParam {
+	case "":
+		codeChallengeMethod = models.SHA256
+	case "plain":
+		codeChallengeMethod = models.Plain
+	case "s256":
+		codeChallengeMethod = models.SHA256
+	case "S256":
+		codeChallengeMethod = models.SHA256
+	default:
+		codeChallengeMethod = models.Unsupported
+	}
+	if codeChallengeMethod == models.Unsupported {
+		return badRequestError("code challenge method is unsupported")
+	}
```

Comment:
nit: preference for the codeChallengeMethod to be stated explicitly rather than default to s256 if it's missing. On the client lib, this should just default to s256 though.

```suggestion
	switch strings.ToLower(codeChallengeMethodParam) {
	case "plain":
		codeChallengeMethod = models.Plain
	case "s256":
		codeChallengeMethod = models.SHA256
	default:
                return badRequestError("code challenge method is unsupported")
	}
```

Reply:
sure, we can do that